### PR TITLE
feat(device): report a device's name, memory and chiplet count

### PR DESCRIPTION
**Pull request by rebel-jonghewk**
## Summary

Adds the RBLN equivalents of `torch.cuda.get_device_properties()` and
`torch.cuda.get_device_name()`:

```python
>>> torch.rbln.get_device_properties(0)
DeviceProperties(name='RBLN-CR13', total_memory=150323855360,
                 memory_per_chiplet=37580963840, num_chiplet=4, npu_count=1)
```

The returned properties describe one **logical** `rbln` device, which may span
multiple physical NPUs:

- `name`: NPU model name
- `total_memory`: total memory across all backing NPUs
- `memory_per_chiplet`: memory available to each chiplet
- `num_chiplet`: chiplets per NPU
- `npu_count`: physical NPUs backing the logical device

`memory_per_chiplet` remains separate from `total_memory` because memory pools
are constrained per chiplet, while capacity checks generally need the aggregate.

## Why the aggregation lives here

[`rbln_get_device_properties()`](https://github.com/RBLN-SW/rebel_compiler/pull/13614)
reports one physical NPU at a time. The logical-to-physical mapping belongs to
`DeviceMappingManager` in torch-rbln, so this layer:

1. resolves a logical device to its physical NPU IDs;
2. queries each NPU through the runtime API;
3. sums `total_memory`; and
4. preserves per-NPU values for `name`, `memory_per_chiplet`, and `num_chiplet`.

An aggregation with different NPU names or chiplet counts raises instead of
returning a misleading single set of properties.

The query does not allocate memory or claim a device. It is therefore safe to
use for capacity discovery before runtime initialization. Calls in
`RBLN_DUMMY_DEVICE` mode raise because no hardware exists to describe.

## Implementation

- Adds `c10::rbln::DeviceProperties` and logical-device aggregation.
- Exposes the properties object and query through pybind.
- Adds the public Python APIs with `int`, `str`, `torch.device`, and omitted
  device arguments.
- Updates the rebel-compiler pin to the merged runtime implementation. The new
  entry point requires ABI 5 and is covered by the existing ABI handshake.
- Adds focused tests for argument handling, current-device behavior, memory
  accounting, multi-NPU aggregation, narrowed visible-device pools, invalid
  indices, and dummy mode.

## Review guide

Suggested review order:

1. `c10/rbln/RBLNFunctions.{h,cpp}` — data model and aggregation semantics
2. `torch_rbln/csrc/rbln/Module.cpp` — Python binding
3. `torch_rbln/device/device.py` — public API and documentation
4. `test/rbln/test_device_properties.py` — edge cases and mapping regressions
5. `constraints-build-dev.txt` — runtime dependency pin

## Related work

- Runtime API: [RBLN-SW/rebel_compiler#13614](https://github.com/RBLN-SW/rebel_compiler/pull/13614)
- Per-chiplet memory reporting: [#191](https://github.com/RBLN-SW/torch-rbln/pull/191)

## Validation

```bash
uv run --no-sync pytest test/rbln/test_device_properties.py
uv run --no-sync python test/run_tests.py --suite=core
uv run --no-sync lintrunner -m origin/main
```

Validated on a CR13 8-card host:

- Targeted device-property tests passed.
- Core suite: 4,293 passed, 20 skipped, 3 xfailed, with one unrelated
  environment-specific failure in
  `test_rbln_runtime_lib.py::TestRblnRuntimeLibRecord::test_the_installed_library_is_under_its_recorded_directory`.
- Lint passed.

The regression tests specifically verify that:

- `total_memory == memory_per_chiplet * num_chiplet * npu_count`;
- aggregating two NPUs doubles only `total_memory` and `npu_count`; and
- an `RBLN_VISIBLE_DEVICES` pool ID is resolved exactly once.

## Design notes

- `num_chiplet` is per NPU. The logical device's total chiplet count is
  `num_chiplet * npu_count`.
- Properties are intentionally not cached. Each call performs one runtime query
  per backing NPU, avoiding stale results when device mappings change between
  processes or tests.

## Type of change

- [x] `feature` — New capability or functionality
- [x] `core` — RBLN backend, device layer, or C++ extension

## Affected modules

- [x] Device
- [x] C++ extension (`torch_rbln/csrc`)

## Checklist

- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] This PR is linked to a corresponding issue — see [Contributing to torch-rbln](https://github.com/RBLN-SW/torch-rbln/blob/main/docs/CONTRIBUTING.md)
- [x] Linting passes — see [Linting](https://github.com/RBLN-SW/torch-rbln/blob/main/docs/LINTING.md)
- [ ] All CI tests pass — see [CI/CD Workflows](https://github.com/RBLN-SW/torch-rbln/blob/main/docs/WORKFLOWS.md)
- [x] The test method and expected result are described
- [ ] Relevant documentation has been updated (if applicable)

### CODE REVIEW

File: test/rbln/test_device_properties.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+import os
+import subprocess
+import sys
+
+import pytest
+import torch
+from torch.testing._internal.common_utils import run_tests, TestCase
+
+from ..utils import requires_physical_devices
```
Comment (rebel-chanheo) [inline test/rbln/test_device_properties.py:15]: nit: `from ..utils import` → `from test.utils import`, matching every other module under `test/rbln`.

**Review by rebel-chanheo — APPROVED** [review]
LGTM with nits
